### PR TITLE
[Qos] [MASIC] supporting Qos changes in PR #5452 on multi-asic

### PR DESCRIPTION
**Pull request by wenyiz2021**
### Description of PR

Qos tests are failing on multi-asic device after PR #5452

Summary:
Fixes # (issue) 
https://github.com/Azure/sonic-mgmt/issues/5539

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [X] 202012

### CODE REVIEW

**Comment by wenyiz2021** [comment]
Tests are passing now on multi-asic devices:
qos/test_qos_sai.py::TestQosSai::testParameter[0] PASSED                                                                                                                           [  1%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[0-xoff_1] PASSED                                                                                                           [  2%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[0-xoff_2] PASSED                                                                                                           [  3%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXonLimit[0-xon_1] SKIPPED                                                                                                            [  4%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXonLimit[0-xon_2] SKIPPED                                                                                                            [  5%]
qos/test_qos_sai.py::TestQosSai::testQosSaiHeadroomPoolSize[0] SKIPPED                                                                                                             [  6%]
qos/test_qos_sai.py::TestQosSai::testQosSaiHeadroomPoolWatermark[0] SKIPPED                                                                                                        [  8%]
qos/test_qos_sai.py::TestQosSai::testQosSaiBufferPoolWatermark[0-wm_buf_pool_lossless] SKIPPED                                                                                     [  9%]
qos/test_qos_sai.py::TestQosSai::testQosSaiBufferPoolWatermark[0-wm_buf_pool_lossy] SKIPPED                                                                                        [ 10%]
qos/test_qos_sai.py::TestQosSai::testQosSaiLossyQueue[0] PASSED                                                                                                                    [ 11%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDscpQueueMapping[0] PASSED                                                                                                              [ 12%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDot1pQueueMapping[0] SKIPPED                                                                                                            [ 13%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDot1pPgMapping[0] SKIPPED                                                                                                               [ 15%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDwrr[0] PASSED                                                                                                                          [ 16%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPgSharedWatermark[0-wm_pg_shared_lossless] PASSED                                                                                       [ 17%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPgSharedWatermark[0-wm_pg_shared_lossy] PASSED                                                                                          [ 18%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPgHeadroomWatermark[0] 
...

**Comment by wenyiz2021** [comment]
================================================ short test summary info =================================================
SKIPPED [8] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:371: Buffer Pool watermark test is disabled
SKIPPED [4] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:561: Dot1p-PG mapping is not supported on t1-64-lag
SKIPPED [4] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:170: PFC Xon Limit test: Need at least 3 destination ports
SKIPPED [4] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:844: DSCP to PG mapping test disabled
SKIPPED [4] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:527: Dot1p-queue mapping is not supported on t1-64-lag
SKIPPED [4] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:306: Headroom pool watermark is not supported
SKIPPED [4] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:229: Headroom pool size not supported
FAILED qos/test_qos_sai.py::TestQosSaiMasic::test_qos_masic_dscp_queue_mapping[4] - Failed: Route 10.0.0.53 doesn't hav...
FAILED qos/test_qos_sai.py::TestQosSaiMasic::test_qos_masic_dscp_queue_mapping[5] - Failed: Route 10.0.0.53 doesn't hav...
=================================== 2 failed, 52 passed, 32 skipped in 6465.40 seconds ===================================

**Comment by wenyiz2021** [comment]
the 2 failed cases were originally failing

**Comment by wenyiz2021** [comment]
failed checks at: tests.macsec.test_macsec.TestControlPlane
To check with Ze Gan

File: tests/common/devices/multi_asic.py [hunk 1]
```diff
@@ -567,3 +567,30 @@ def get_internal_bgp_peers(self):
...
+        """This function copy from host to ALL asics"""
+        duthost = self.sonichost
+        if duthost.is_multi_asic:
+            for n in range(duthost.facts['num_asic']):
+                container = container_name + str(n)
+                self.shell("sudo docker cp {} {}:{}".format(src, container, dst))
+        else:
+            self.shell("sudo docker cp {} {}:{}".format(src, container_name, dst))
+
+    def docker_copy_from_asic0(self, container_name, src, dst):
```
Comment (neethajohn) [inline tests/common/devices/multi_asic.py:591]: suggest to rename this to "docker_copy_from_asic" and change the func signature to have an asic parameter and have 0 as the default.  In that way, in future if there is a need to copy from other asics, this function can be used
docker_copy_from_asic(self, container_name, src, dest, asic=0):

**Review by neethajohn — COMMENTED** [review]
Please verify that the testcases work fine on a single asic device as well

**Comment by wenyiz2021** [comment]
Merged PR: https://github.com/Azure/sonic-mgmt/pull/5319 to see if help multi-asic qos tests to pass.

**Comment by wenyiz2021** [comment]
> Please verify that the testcases work fine on a single asic device as well

@neethajohn 
Here is the single asic qos run results:
qos/test_qos_sai.py::TestQosSai::testParameter[None] PASSED                                                                                                                                                                                        [  4%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[None-xoff_1] PASSED                                                                                                                                                                        [  9%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[None-xoff_2] PASSED                                                                                                                                                                        [ 13%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXonLimit[None-xon_1] PASSED                                                                                                                                                                          [ 18%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXonLimit[None-xon_2] PASSED                                                                                                                                                                          [ 22%]
qos/test_qos_sai.py::TestQosSai::testQosSaiHeadroomPoolSize[None] XPASS                                                                                                                                                                            [ 27%]
qos/test_qos_sai.py::TestQosSai::testQosSaiHeadroomPoolWatermark[None] XPASS                                                                                                                                                                       [ 31%]
qos/test_qos_sai.py::TestQosSai::testQosSaiBufferPoolWatermark[None-wm_buf_pool_lossless] SKIPPED                                                                                                                                                  [ 36%]
qos/test_qos_sai.py::TestQosSai::testQosSaiBufferPoolWatermark[None-wm_buf_pool_lossy] SKIPPED                                                                                                                                                     [ 40%]
qos/test_qos_sai.py::TestQosSai::testQosSaiLossyQueue[None] PASSED                                                                                                                                                                                 [ 45%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDscpQueueMapping[None] PASSED                                                                                                                                                                           [ 50%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDot1pQueueMapping[None] SKIPPED                                                                                                                                                                         [ 54%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDot1pPgMapping[None] SKIPPED                                                                                                                                                                            [ 59%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDwrr[None] PASSED                                                                                                                                                                                       [ 63%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPgSharedWatermark[None-wm_pg_shared_lossless] PASSED                                                                                                                                                    [ 68%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPgSharedWatermark[None-wm_pg_shared_lossy] XPASS                                                                                                                                                        [ 72%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPgHeadroomWatermark[None] PASSED                                                                                                                                                                        [ 77%]
qos/test_qos_sai.py::TestQosSai::testQosSaiQSharedWatermark[None-wm_q_shared_lossless] PASSED                                                                                                                                                      [ 81%]
qos/test_qos_sai.py::TestQosSai::testQosSaiQSharedWatermark[None-wm_q_shared_lossy] PASSED                                                                                                                                                         [ 86%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDscpToPgMapping[None] SKIPPED                                                                                                                                                                           [ 90%]
qos/test_qos_sai.py::TestQosSai::testQosSaiDwrrWeightChange[None] PASSED                                                                                                                                                                           [ 95%]
qos/test_qos_sai.py::TestQosSaiMasic::test_qos_masic_dscp_queue_mapping[None] SKIPPED                                                                                                                                                              [100%]

--------------------------------------------------------------------------------------------- generated xml file: /var/src/sonic-mgmt-int/tests/logs/tr.xml ----------------------------------------------------------------------------------------------
================================================================================================================ short test summary info =================================================================================================================
SKIPPED [1] /var/src/sonic-mgmt-int/tests/qos/qos_sai_base.py:1523: unsupported topology t0
SKIPPED [1] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:561: Dot1p-PG mapping is not supported on t0
SKIPPED [2] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:371: Buffer Pool watermark test is disabled
SKIPPED [1] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:844: DSCP to PG mapping test disabled
SKIPPED [1] /var/src/sonic-mgmt-int/tests/qos/test_qos_sai.py:527: Dot1p-queue mapping is not supported on t0
XPASS qos/test_qos_sai.py::TestQosSai::testQosSaiHeadroomPoolSize[None] Image issue on Arista platforms
XPASS qos/test_qos_sai.py::TestQosSai::testQosSaiHeadroomPoolWatermark[None] Image issue on Arista platforms
XPASS qos/test_qos_sai.py::TestQosSai::testQosSaiPgSharedWatermark[None-wm_pg_shared_lossy] Image issue on Arista platforms
=================================================================================================== 13 passed, 6 skipped, 3 xpassed in 2831.96 seconds ===================================================================================================

**Review by neethajohn — APPROVED** [review]

**Comment by wenyiz2021** [comment]
The PR check fail at arp/test_arp_dualtor.py, took a dualtor t0 to run this test before & after the change in this PR, it succeed:
arp/test_arp_dualtor.py::test_proxy_arp[v4-str2-7050cx3-acs-10-None] PASSED                                                                                                        [  9%]
arp/test_arp_dualtor.py::test_proxy_arp[v6-str2-7050cx3-acs-10-None] PASSED                                                                                                        [ 18%]
metadata-scripts/test_metadata_upgrade_path.py::test_cancelled_upgrade_path[warm-str2-7050cx3-acs-10] SKIPPED                                                                      [ 27%]
metadata-scripts/test_metadata_upgrade_path.py::test_upgrade_path[warm-str2-7050cx3-acs-10] SKIPPED                                                                                [ 36%]
metadata-scripts/test_metadata_upgrade_path.py::test_warm_upgrade_sad_path[warm-sad-str2-7050cx3-acs-10] SKIPPED                                                                   [ 45%]
metadata-scripts/test_metadata_upgrade_path.py::test_warm_upgrade_sad_path[warm-multi_sad-str2-7050cx3-acs-10] SKIPPED                                                             [ 54%]
metadata-scripts/test_metadata_upgrade_path.py::test_warm_upgrade_sad_path[warm-sad_bgp-str2-7050cx3-acs-10] SKIPPED                                                               [ 63%]
metadata-scripts/test_metadata_upgrade_path.py::test_warm_upgrade_sad_path[warm-sad_lag_member-str2-7050cx3-acs-10] SKIPPED                                                        [ 72%]
metadata-scripts/test_metadata_upgrade_path.py::test_warm_upgrade_sad_path[warm-sad_lag-str2-7050cx3-acs-10] SKIPPED                                                               [ 81%]
metadata-scripts/test_metadata_upgrade_path.py::test_warm_upgrade_sad_path[warm-sad_vlan_port-str2-7050cx3-acs-10] SKIPPED                                                         [ 90%]
metadata-scripts/test_metadata_upgrade_path.py::test_warm_upgrade_sad_path[warm-sad_inboot-str2-7050cx3-acs-10] SKIPPED                                                            [100%]
========================================================================= 2 passed, 9 skipped in 148.44 seconds ==========================================================================

**Pull request merged**